File: admin-react/src/lib/random/studentDetail.js
import { faker } from "@faker-js/faker";
faker.setLocale("zh_CN");
import { getDate } from "../../util/useDate";

export function randomStudentDetail() {
  return {
    gender: faker.name.sexType() === "male" ? "男" : "女",
    birthDate: getDate(
      faker.date.between("1980-01-01T00:00:00.000Z", "2003-01-01T00:00:00.000Z")
    ),
    classes: faker.helpers.arrayElement([
      "19计科本5",
      "20计科本5",
      "21计科本5",
    ]),
    grade: faker.date.past().getFullYear(),
    labId: "0",
    phone: faker.phone.number(),
    email: faker.internet.email(),
    wechat: faker.word.adjective(),
    qq: faker.word.adjective(),
    description: faker.lorem.paragraph(),
    studyNum: faker.datatype.number({
      min: 0,
      max: 100,
    }),
    sportNum: faker.datatype.number({
      min: 0,
      max: 100,
    }),
    communicationNum: faker.datatype.number({
      min: 0,
      max: 100,
    }),
    competitionNum: faker.datatype.number({
      min: 0,
      max: 100,
    }),
    licenseNum: faker.datatype.number({
      min: 0,
      max: 100,
    }),
    handsNum: faker.datatype.number({
      min: 0,
      max: 100,
    }),
  };
}

File: admin-springboot/src/main/java/hbsm/zhxy/studentdetail/StudentDetail.java
package hbsm.zhxy.studentdetail;

import hbsm.zhxy.student.Student;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import lombok.Data;

@Entity
@Data
public class StudentDetail {
    @Id
    private String id;

    @OneToOne
    @JoinColumn(name = "id")
    private Student student;

    private String gender;
    private String phone;
    private String email;
    private String wechat;
    private String qq;
    private String birthDate;
    private String description;
    private int studyNum;
    private int sportNum;
    private int communicationNum;
    private int competitionNum;
    private int licenseNum;
    private int handsNum;
}
